src/Stan/Toml.hs

```diff
@@ -32,25 +34,41 @@ import qualified Toml
 
 
 getTomlConfig :: IO PartialConfig
-getTomlConfig = do
-    file <- defaultConfigFile
-    isFile <- doesFileExist file
-    if isFile
-    then do
-        infoMessage $ "Configurations from " <> toText defaultTomlFile <> " will be used."
-        Toml.decodeFile configCodec file
-    else let e = fiasco "Configurations file doesn't exist" in
-        pure $ ConfigP e
+getTomlConfig = defaultCurConfigFile >>= readToml >>= \case
+    Result _ r -> pure r
+    resCur -> defaultHomeConfigFile >>= readToml >>= \ resHome ->
+        pure $ case resCur <> resHome of
+            Fiasco f     -> ConfigP $ Fiasco f
+            Result _ res -> res
```

Comment:
This is a cool idea! Looks like something where `trial` can shine. We need a function that will take a list of actions and perform them sequentially until first `Result` and also combine all histories. I'm thinking about the type signature:

```haskell
trials :: Monad m => [m a] -> (a -> m (Trial e b)) -> m (Trial e b)
```

Reply:
Nice idea! Should create an issue in `trial` :wink: